src/ducks/settings/PinSettings.jsx

```diff
@@ -1,48 +1,117 @@
 import React from 'react'
+import { queryConnect, withMutations } from 'cozy-client'
+import compose from 'lodash/flowRight'
 
-import Button from 'cozy-ui/react/Button'
 import { translate } from 'cozy-ui/react'
+import Icon from 'cozy-ui/react/Icon'
+import ToggleRow from 'ducks/settings/ToggleRow'
 
 import PinEditView from 'ducks/pin/PinEditView'
+import PinAuth from 'ducks/pin/PinAuth'
+import PinWrapper from 'ducks/pin/PinWrapper'
+import PinButton from 'ducks/pin/PinButton'
+import { pinSetting } from 'ducks/pin/queries'
+
+console.clear()
+
+const saveScroll = () => {
+  let scrollLeft = document.body.scrollLeft
+  let scrollTop = document.body.scrollTop
+  return () => {
+    document.body.scrollTo(scrollLeft, scrollTop)
+  }
+}
 
 class PinSettings extends React.Component {
   state = {
-    configuring: false
+    togglingOn: false,
+    togglingOff: false
   }
 
   constructor(props) {
     super(props)
-    this.handleConfigure = this.handleConfigure.bind(this)
     this.handleExit = this.handleExit.bind(this)
     this.handlePinSaved = this.handlePinSaved.bind(this)
+    this.handleToggle = this.handleToggle.bind(this)
+    this.handlePinDeactivated = this.handlePinDeactivated.bind(this)
+    this.handleTogglingOffCancel = this.handleTogglingOffCancel.bind(this)
+  }
+
+  getPinDoc() {
+    const { pinSetting } = this.props
+    return (pinSetting && pinSetting.data) || null
   }
 
-  handleConfigure() {
-    this.setState({ configuring: true })
+  handleToggle() {
+    const pinDoc = this.getPinDoc()
+    this.restoreScroll = saveScroll(document.body)
```

Comment:
The above `saveScroll` function doesn't take a parameter, so `document.body` seems to be unnecessary